### PR TITLE
ARROW-8121: [Java] Enhance code style checking for Java code (add spaces after commas, semi-colons and type casts)

### PR DESCRIPTION
**Pull request by liyafan82**
This is in response to a discussion in https://github.com/apache/arrow/pull/6039#discussion_r375161992

We found the current style checking for Java code is not sufficient. So we want to enhace it in a series of "small" steps, in order to avoid having to change too many files at once.

In this issue, we add spaces after commas, semi-colons and type casts.

### CODE REVIEW

**Review by rymurr — COMMENTED** [review]
Aside from the 2x double spaces lgtm

File: java/gandiva/src/test/java/org/apache/arrow/gandiva/evaluator/ProjectorDecimalTest.java
Comment (liyafan82) [reply java/gandiva/src/test/java/org/apache/arrow/gandiva/evaluator/ProjectorDecimalTest.java]: @rymurr Thanks a lot for your careful review. The problems are resolved. I guess my eyes were bad.

**Comment by rymurr** [comment]
Looks good to me! :+1:

**Review by rymurr — APPROVED** [review]

**Comment by emkornfield** [comment]
@liyafan82 would you minds rebasing this?

**Comment by liyafan82** [comment]
> @liyafan82 would you minds rebasing this?

Done. Thank you.

**Comment by emkornfield** [comment]
@liyafan82 sorry for the delay would you mind rebasing once more?  I'll merge later tonight if I'm online or first thing tomorrow, so you won't need to do it again.

**Comment by liyafan82** [comment]
> @liyafan82 sorry for the delay would you mind rebasing once more? I'll merge later tonight if I'm online or first thing tomorrow, so you won't need to do it again.

Rebased. Thanks for your effort.

**Pull request closed without merging**
